### ISSUE TITLE
Feature Request: Add Official Supabase Support

### ISSUE THREAD

**Issue by About7Deaths** [logto-io/logto#1797]
### What problem did you meet?

It is unclear if Supabase is properly / fully supported. The authentication flow I have in mind is using Logto as the identity and authentication server for the login page of my application, which would then redirect to an application dashboard populated by the various Supabase database features (besides their built-in authentication).

### Describe what you'd like Logto to have

I am unsure if changes only need to be made to Supabase, or additionally to Logto. To facilitate further discussion of this problem, I created an issue on the Superbase repository requesting that they add support for Logto OIDC - see [Issue 407 Reference](https://github.com/supabase/gotrue/issues/881).

**Comment by gao-sun** [comment]
thank you! really cool idea

**Comment by baderdean** [comment]
Hello,

is there some updates to this feature request? Supabase uses Gotrue yet the IAM part is very basic. I rather prefer a proper solution like Logto for a better user experience.

**Comment by gao-sun** [comment]
@baderdean yes! we're looking at it. we are new to supabase and its integration process takes more time than we expected. will keep you posted.

**Comment by gao-sun** [comment]
After some investigation, before Logto has a "custom token signing" feature, I think it's pretty close to the Auth0 integration:

- https://supabase.com/docs/guides/integrations/auth0
- https://auth0.com/blog/using-nextjs-and-auth0-with-supabase/

Since:

- For now Logto cannot be used as a third-party identity provider
- For the first-party integration, you also need to manually sign the JWT after user signed in

Replace the Auth0 part with Logto and it'll be good I suppose. @wangsijie would you like to double check when you have time? thanks.

**Comment by wangsijie** [comment]
Replace the Auth0 part with Logto will work! But I am not sure whether this can solve your problem. @About7Deaths 
In this way, Logto will take place the whole built-in authentication provided by Supabase, as for authorization, some fetures like "Row Level Security" will still be available, but the other parts like "Managing User Data" may not work.

**Comment by baderdean** [comment]
Too bad it has been closed as not planned. We're going for GoTrue then.

**Comment by gao-sun** [comment]
> Too bad it has been closed as not planned. We're going for GoTrue then.

ah the auto close is unexpected

**Comment by wangsijie** [comment]
@baderdean @About7Deaths After a further investigation, I am sure that @gao-sun's conclusion works, can it solve your problem?

**Comment by baderdean** [comment]
We're using Supabase Gotrue because it was easier but still unhappy about many stuff like: no default UI, no nice email templates, no ability to have multi-social login account linked to the auth account, no ability to get additional credentials besides the basic ones during auth (for example accessing the contact lists)

**Comment by Rany0101** [comment]
This feature is supported. Refer to: https://docs.logto.io/quick-starts/supabase
